app/lib/task/backend.dart

```diff
@@ -686,13 +686,14 @@ class TaskBackend {
       final key = PackageState.createKey(_db, runtimeVersion, package);
       final state = await tx.lookupOrNull<PackageState>(key);
       if (state == null || state.versions![version] == null) {
-        throw NotFoundException.resource('$package/$version');
+        throw NotAcceptableException(
```

Comment:
406 is probably not the right status code to use here.

We can use 404, because if the resources we're accessing (as indicated by URL) doesn't exist anymore.
We could also choose a generic 400 error.

I'm not sure that 409 is accurate because it's indicates that the client could do something to resolve the issue.

------------

I'm actually pretty sure the correct answer here is 404. Because someone could making a POST request to `/api/tasks/$package/$version/finished` with a `package` and `version` that simply don't exist.

We could add a custom error message to `NotFoundException`. But if `package` and `version` combination don't exist, then this is probably the correct message.

It's probably a better option to check datastore to see if the given package and version exists.
If they do, but they are not scheduled for analysis in `PackageState` we could return `400` instead. But then I'd suggest we make a custom `400` with its own `code`:
```dart
/// Thrown when results are being uploaded for a package/version where
/// analysis is not pending.
class AnalysisNotPendingException extends ResponseException {
  AnalysisNotPendingException._(String message)
      : super._(
          400,
          'AnalysisNotPending',
          message,
        );
```

Then we can parse the JSON and take out the `code` property.

------------

But I'm also inclined to suggest that what we could do is:
 * Handle 404 in `pub_worker` and just silence it.
   (this seems a bit risky, we could hiding future issues)
 * Investigate if we can avoid removing `PackageState` entries that are actively pending analysis (where an instance currently running).
   * This is probably difficult
   * We'd have to rely on timestamps to know if an instance is likely alive.
   * We'd need to review `PackageState` other methods, etc. as we are possibly changing invariants.
  (This might not be something we'll want to do, just considering the shear complexity)
 * Handle `404` in `pub_worker` and fetch version-listing from pub.dev and review the `published` date-time on the `latest` entry.
   * If there is a newer version not in the list of versions we've been told to analyze, this could be an indicator of a 404 error or 401 error not being important.
   * We could also just say that if `published` is recent, then `404` and `401` are reasonable errors to get?